waspc/src/Wasp/Generator/AuthProviders/OAuth.hs

```diff
@@ -0,0 +1,64 @@
+module Wasp.Generator.AuthProviders.OAuth
+  ( frontendLoginUrl,
+    serverLoginUrl,
+    serverOauthRedirectHandlerUrl,
+    slug,
+    displayName,
+    logoFileName,
+    passportDependency,
+    scopeStr,
+    clientIdEnvVarName,
+    clientSecretEnvVarName,
+    OAuthAuthInfo (..),
+  )
+where
+
+import Data.Char (toUpper)
+import Data.List (intercalate)
+import StrongPath (File', Path', Rel')
+import Wasp.AppSpec.App.Dependency (Dependency)
+
+data OAuthAuthInfo = OAuthAuthInfo
+  { _slug :: String,
+    _displayName :: String,
+    _requiredScope :: OAuthScope,
+    _logoFileName :: Path' Rel' File',
+    _passportDependency :: Dependency
+  }
+
+type OAuthScope = [String]
+
+slug :: OAuthAuthInfo -> String
+slug = _slug
+
+displayName :: OAuthAuthInfo -> String
+displayName = _displayName
+
+logoFileName :: OAuthAuthInfo -> Path' Rel' File'
+logoFileName = _logoFileName
+
+clientIdEnvVarName :: OAuthAuthInfo -> String
+clientIdEnvVarName oai = upperCaseSlug oai ++ "_CLIENT_ID"
+
+clientSecretEnvVarName :: OAuthAuthInfo -> String
+clientSecretEnvVarName oai = upperCaseSlug oai ++ "_CLIENT_SECRET"
+
+upperCaseSlug :: OAuthAuthInfo -> String
+upperCaseSlug oai = map toUpper (_slug oai)
+
+scopeStr :: OAuthAuthInfo -> String
```

Comment:
Generates the string used in JS e.g. `["profile"]` list in Haskell becomes `"[\"profile\"]"` string which can be outputted in JS code verbatim.

Reply:
I think this is an OK compromise for now 👍🏻 All of the Haskell -> JS stuff has some warts here and there already.

Reply:
Compromise may be a bit strong, but just making a compound type into a primitive feels off somehow. :) But it's fine since it's localized and the net result is the same! Not worth the hassle of passing an array of strings just to do what we are now 👍🏻

Reply:
Ah got it, makes sense!